features/device-config/retrieval/src/main/java/org/opennms/features/deviceconfig/retrieval/impl/RetrieverImpl.java

```diff
@@ -61,16 +63,28 @@
 
     private static Logger LOG = LoggerFactory.getLogger(RetrieverImpl.class);
 
-    private static String FILENAME = "filename";
+    private static String SCRIPT_VAR_FILENAME = "filename";
+    private static String SCRIPT_VAR_TFTP_SERVER_IP = "tftpServerIp";
+    private static String SCRIPT_VAR_TFTP_SERVER_PORT = "tftpServerPort";
+    private static String SCRIPT_VAR_CONFIG_TYPE = "configType";
 
     private final SshScriptingService sshScriptingService;
     private final TftpServer tftpServer;
     private final ExecutorService executor;
+    private final String tftpServerIp;
 
-    public RetrieverImpl(SshScriptingService sshScriptingService, TftpServer tftpServer) {
+    public RetrieverImpl(SshScriptingService sshScriptingService, TftpServer tftpServer) throws Exception {
         this.sshScriptingService = sshScriptingService;
         this.tftpServer = tftpServer;
         this.executor = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("device-config-retriever-%d").build());
+        this.tftpServerIp = determineIp();
+    }
+
+    private String determineIp() throws Exception {
+        try(final DatagramSocket socket = new DatagramSocket()) {
+            socket.connect(InetAddress.getByName("8.8.8.8"), 10002);
```

Comment:
Hrm, yeah, I'm not sure this is a good idea as implemented, we have plenty of installs running in locked down environments, requiring access to resolving an outside hostname should not be assumed, especially one specifically run by Google.  :D

(I realize you don't have to be able to _connect_ to that host, just resolve it, but even _that_ should not be assumed.)

Figuring out your "real" IP is fraught with peril, so while most of the time this would work, IMHO it is dangerous to _assume_ that this will work.  At the very least, can we make it possible to explicitly configure the IP to connect to in metadata, and fall back to this behavior if it's not set? That at least leaves an out for weird environments.

I assume that's easier than trying to ask the device you connect to what it thinks your IP is (like `who -m` on linux, for example).

Reply:
I think the operation is not critical. It is like a fire and forget (send an UDP package) and then ask what was the used sender address.

Reply:
I don't think so. I copied these lines form Stackoverflow (cf. https://stackoverflow.com/questions/9481865/getting-the-ip-address-of-the-current-machine-using-java). However, I am not an expert in those low-level mechanics.

Reply:
OK, just did a test, and I missed this the first time around reading the java docs:

>  The host name can either be a machine name, such as "java.sun.com", or a textual representation of its IP address. If a literal IP address is supplied, **_only the validity of the address format is checked_**.

So it does _not_ rely on actually resolving the address, since it is parseable as an IP address.  You can literally use any IP you want in here, so maybe it would be more clear that we don't care what it is by using `InetAddressUtils.UNPINGABLE_ADDRESS`.

```suggestion
            socket.connect(InetAddressUtils.UNPINGABLE_ADDRESS, 10002);
```

Reply:
Tests by @RangerRick showed that it works even in case that DNS can not resolve anything.